app/uk/gov/hmrc/transitmovementsrouter/services/RoutingService.scala

```diff
@@ -0,0 +1,126 @@
+import uk.gov.hmrc.transitmovementsrouter.services.ParseError.Unknown
+import uk.gov.hmrc.transitmovementsrouter.services.error.RoutingError
+
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[RoutingServiceImpl])
+trait RoutingService {
+
+  def submitDeclaration(
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    payload: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier): Future[Either[ParseError, Int]]
+}
+
+class RoutingServiceImpl @Inject() (messageConnectorProvider: MessageConnectorProvider)(implicit materializer: Materializer)
+    extends RoutingService
+    with XmlParsingServiceHelpers
+    with Logging {
+
+  val office = Sink.head[ParseResult[OfficeOfDeparture]]
+
+  private val officeOfDepartureSink: Sink[ByteString, Future[ParseResult[OfficeOfDeparture]]] = Sink.fromGraph(
+    GraphDSL.createGraph(office) {
+      implicit builder => officeShape =>
+        import GraphDSL.Implicits._
+
+        val xmlParsing: FlowShape[ByteString, ParseEvent]                            = builder.add(XmlParsing.parser)
+        val officeOfDeparture: FlowShape[ParseEvent, ParseResult[OfficeOfDeparture]] = builder.add(XmlParser.officeOfDepartureExtractor)
+        xmlParsing ~> officeOfDeparture ~> officeShape
+
+        SinkShape(xmlParsing.in)
+    }
+  )
+
+  def buildMessage(messageId: MessageId): Flow[ByteString, ByteString, NotUsed] =
+    Flow.fromGraph(
+      GraphDSL.create() {
+        implicit builder =>
+          import GraphDSL.Implicits._
+
+          val xmlParsing: FlowShape[ByteString, ParseEvent]         = builder.add(XmlParsing.parser)
+          val insertSenderWriter: FlowShape[ParseEvent, ParseEvent] = builder.add(XmlParser.messageSenderWriter(messageId))
+          val toByteString                                          = builder.add(XmlWriting.writer)
+          xmlParsing ~> insertSenderWriter ~> toByteString
+
+          FlowShape(xmlParsing.in, toByteString.out)
+      }
+    )
+
+  override def submitDeclaration(
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    payload: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier): Future[Either[ParseError, Int]] = {
+
+    implicit val officeOfDeparture = payload.toMat(officeOfDepartureSink)(Keep.right).run()
+    implicit val updatedPayload    = payload.via(buildMessage(messageId))
+    implicit val messageSender     = MessageSender(messageId.value)
+
+    post(officeOfDeparture)
+  }
+
+  //TODO:  Not quite sure what we return here
+  private def post(
+    office: Future[ParseResult[OfficeOfDeparture]]
+  )(implicit messageSender: MessageSender, body: Source[ByteString, _], hc: HeaderCarrier): Future[Either[ParseError, Int]] = {
+    import concurrent.ExecutionContext.Implicits.global
+    office.flatMap {
+      case Right(office) =>
+        postToEis(office).map {
+          case Right(_)    => Right(play.api.http.Status.ACCEPTED)
```

Comment:
Should we return unit instead of the status code? If it succeeds the status code returned will always be ACCEPTED, so that's something that could be derived in the controller if it gets a Right

Reply:
Agreed - will return Unit.